pom.xml

```diff
@@ -355,18 +355,6 @@
                 </plugins>
             </build>
         </profile>
-
-        <profile>
-            <id>spoonSnapshot</id>
-            <repositories>
-                <repository>
-                    <id>ow2.org-snapshot</id>
-                    <name>Maven Repository for Spoon Snapshots</name>
-                    <url>https://repository.ow2.org/nexus/content/repositories/snapshots/</url>
-                    <snapshots><enabled>true</enabled></snapshots>
-                </repository>
-            </repositories>
-        </profile>
```

Comment:
I removed this XML config because it was not working for fetching SNAPSHOTs of Spoon when I was trying to debug Sorald using it. For using SNAPSHOT version, we can refer how it is done in [`diffmin`](https://github.com/SpoonLabs/diffmin/pull/90/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8L55-L61) because I know it works there.

Reply:
You don't want to do it like it is in diffmin, because that adds a repository to anything that depends on diffmin. In general, one should avoid adding repositories in pom files, [here's a pretty good article about it](https://blog.sonatype.com/2009/02/why-putting-repositories-in-your-poms-is-a-bad-idea/).

Putting the repository in a profile means that it won't be included by default in dependent libraries/applications, which is desirable. As a concrete example of why it's bad to put repositories in pom files (without putting them in custom profiles): the old snapshot repository we used for Spoon was in its pom file, and it was pretty unreliable and went down every now and then. And every time it did, it wasn't possible to build Spoon _or anything depending on Spoon_, as that repository couldn't be reached. Even though nothing was fetched from it.

This works perfectly if you activate `spoonSnapshot` profile, e.g. `mvn -P spoonSnapshot compile`, but by default the repository isn't included (which is what we want).

Reply:
I did not know this. Thanks!

I was about to argue that it is still safe to enclose SNAPSHOT URLs in `repositories` however, sorald is also used as a dependency in some other projects, eg. repairnator. So I get your point. I will revert this change whenever we need the SNAPSHOT version next time.

Reply:
> I will revert this change whenever we need the SNAPSHOT version next time.

@algomaster99 Do you think we should open an issue not to forget this? Or should we do it right now?

Reply:
I don't think so opening an issue is necessary because it is not needed and whenever we want to work with the SNAPSHOT version again, we will know what we have to do. :)

Reply:
Fine :)